packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1299,11 +1299,7 @@ function dispatchAction<S, A>(
       // This is the first update. Create a circular list.
       update.next = update;
     } else {
-      const first = pending.next;
-      if (first !== null) {
-        // Still circular.
-        update.next = first;
```

Comment:
What's this about, we need this for it to stay circular, no?

Reply:
All I did was remove the `first !== null` check. Which is unnecessary because `update.next` is never null.

Reply:
I think the diff makes it look like I deleted the assignment but really all I deleted was the type check that surrounds it.

Reply:
Actually, to rephrase, it's not really a type check that I deleted. In the old implementation, the queue was only sometimes circular. It would eventually "unravel" to linear so that the old nodes could be garbage collected. But in the new implementation, the pending queue is *always* circular. So you don't need to check for if it's circular or not.